workflow/executor/agent.go

```diff
@@ -260,18 +260,35 @@ func (ae *AgentExecutor) executeHTTPTemplate(ctx context.Context, tmpl wfv1.Temp
 			message = fmt.Sprintf("received non-2xx response code: %d", response.StatusCode)
 		}
 	} else {
-		evalScope := map[string]interface{}{
-			"request": map[string]interface{}{
-				"method":  tmpl.HTTP.Method,
-				"url":     tmpl.HTTP.URL,
-				"body":    tmpl.HTTP.Body,
-				"headers": tmpl.HTTP.Headers.ToHeader(),
-			},
-			"response": map[string]interface{}{
-				"statusCode": response.StatusCode,
-				"body":       string(bodyBytes),
-				"headers":    response.Header,
-			},
+		var evalScope map[string]interface{}
+		if tmpl.HTTP.BodyFrom != nil {
+			evalScope = map[string]interface{}{
+				"request": map[string]interface{}{
+					"method":  tmpl.HTTP.Method,
+					"url":     tmpl.HTTP.URL,
+					"body":    string(tmpl.HTTP.BodyFrom.Bytes),
+					"headers": tmpl.HTTP.Headers.ToHeader(),
+				},
+				"response": map[string]interface{}{
+					"statusCode": response.StatusCode,
+					"body":       string(bodyBytes),
+					"headers":    response.Header,
+				},
+			}
+		} else {
```

Comment:
Why are we duplicating this block twice if the only difference is the contents of `"body"`? Why not just have a statement setting `"body"` appropriately based on `tmpl.HTTP.BodyFrom`?